### PR TITLE
Formatter refactoring

### PR DESCRIPTION
**Pull request by tobiasdiez**
Refactor the regular expression-based formatter. Previously, the regex was sent through a static `setRegex` method, now it gets passed as a normal constructor argument.

----

- [ ] Change in CHANGELOG.md described
- [ ] Tests created for changes
- [ ] Manually tested changed features in running JabRef
- [ ] Screenshots added in PR description (for bigger UI changes)
- [ ] Ensured that [the git commit message is a good one](https://github.com/joelparkerhenderson/git_commit_message)
- [ ] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)

### CODE REVIEW

File: src/main/java/org/jabref/logic/formatter/Formatters.java [hunk 1]
```diff
@@ -31,83 +31,63 @@
...
+        if (modifier.startsWith(RegexFormatter.KEY)) {
+            String regex = modifier.substring(RegexFormatter.KEY.length());
+            return Optional.of(new RegexFormatter(regex));
         } else {
-            formatter = all.stream().filter(f -> f.getKey().equals(modifier)).findAny();
-        }
-        if (formatter.isPresent()) {
-            return formatter;
+            Optional<Formatter> formatter = getAll().stream().filter(f -> f.getKey().equals(modifier)).findAny();
+            if (formatter.isPresent()) {
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/logic/formatter/Formatters.java]: That does not make any sense does it?
The method returns an optional. So you can directly return formatter.

File: src/test/java/org/jabref/logic/formatter/FormatterTest.java [hunk 2]
```diff
@@ -38,102 +38,101 @@
...
+                new LatexToUnicodeFormatter(),
+                new LowerCaseFormatter(),
+                new MinifyNameListFormatter(),
+                new NormalizeDateFormatter(),
+                new NormalizeMonthFormatter(),
+                new NormalizeNamesFormatter(),
+                new NormalizePagesFormatter(),
+                new OrdinalsToSuperscriptFormatter(),
+                new ProtectTermsFormatter(protectedTermsLoader),
+               new RegexFormatter("(\" \",\"-\")"),
```
Comment (Siedlerchr) [inline src/test/java/org/jabref/logic/formatter/FormatterTest.java]: I don't want to do nitpkicking but that Regex Formatter indentation is clear off

File: src/test/java/org/jabref/logic/formatter/FormatterTest.java [hunk 3]
```diff
@@ -38,102 +38,101 @@
...
+                new UnitsToLatexFormatter(),
+                new UpperCaseFormatter());
 
+        // @formatter:on
     }
 
     @ParameterizedTest
     @MethodSource("getFormatters")
-    public void getNameReturnsNotNull(Formatter formatter) {
+    void getNameReturnsNotNull(Formatter formatter) {
```
Comment (Siedlerchr) [inline src/test/java/org/jabref/logic/formatter/FormatterTest.java]: Why do you remove the public?
Codacy or whatever will complain that you should always use explicit scoping modifiers. I see no harm in using public here.  I don't know if this even has to be public to be accessible from gradle

File: src/test/java/org/jabref/logic/formatter/bibtexfields/RegexFormatterTest.java [hunk 4]
```diff
@@ -1,64 +1,55 @@
...
-import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
 /**
  * Tests in addition to the general tests from {@link org.jabref.logic.formatter.FormatterTest}
  */
-public class RegexFormatterTest {
+class RegexFormatterTest {
```
Comment (Siedlerchr) [inline src/test/java/org/jabref/logic/formatter/bibtexfields/RegexFormatterTest.java]: Same as above

File: src/main/java/org/jabref/logic/formatter/Formatters.java (hunk 1 above)
Comment (tobiasdiez) [reply src/main/java/org/jabref/logic/formatter/Formatters.java]: If no formatter is found, then we still want to try the switch at the end. I now reordered the code a bit and it should be clearer now. Good point.

File: src/test/java/org/jabref/logic/formatter/FormatterTest.java (hunk 2 above)
Comment (tobiasdiez) [reply src/test/java/org/jabref/logic/formatter/FormatterTest.java]: Yeah, no idea why there were these `@formatter` comments that prevented an automatic code formatting...

File: src/test/java/org/jabref/logic/formatter/FormatterTest.java (hunk 3 above)
Comment (tobiasdiez) [reply src/test/java/org/jabref/logic/formatter/FormatterTest.java]: It's the new convention with JUnit 5 that test classes and their methods are package private. IntellJ even complains if they are marked with public.

Comment (Siedlerchr) [reply src/test/java/org/jabref/logic/formatter/FormatterTest.java]: Ah okay, that was new to me

**Pull request merged**